pkg/util/cryptomaterial/signers_test.go

```diff
@@ -204,3 +203,166 @@ func pemToCert(t *testing.T, certPEM []byte) *x509.Certificate {
 func breakTestCertPath(testPath string) []string {
 	return strings.Split(testPath, "/")
 }
+
+func TestCertificateChains_WalkChains(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	testChains, err := NewCertificateChains(
+		NewCertificateSigner("test-signer1", filepath.Join(tmpDir, "test-signer1"), 1).
+			WithClientCertificates(&ClientCertificateSigningRequestInfo{
+				CertificateSigningRequestInfo: CertificateSigningRequestInfo{
+					Name:         "test-client1",
+					ValidityDays: 1,
+				},
+				UserInfo: &user.DefaultInfo{Name: "test-user", Groups: []string{"test-group1", "test-group2"}},
+			},
+				&ClientCertificateSigningRequestInfo{
+					CertificateSigningRequestInfo: CertificateSigningRequestInfo{
+						Name:         "test-client2",
+						ValidityDays: 1,
+					},
+					UserInfo: &user.DefaultInfo{Name: "test-user2"},
+				},
+			).WithServingCertificates(&ServingCertificateSigningRequestInfo{
+			CertificateSigningRequestInfo: CertificateSigningRequestInfo{
+				Name:         "test-signer1-server1",
+				ValidityDays: 1,
+			},
+			Hostnames: []string{"behind.the.wardrobe.door"},
+		}),
+		NewCertificateSigner("test-signer2", filepath.Join(tmpDir, "test-signer2"), 1).
+			WithServingCertificates(&ServingCertificateSigningRequestInfo{
+				CertificateSigningRequestInfo: CertificateSigningRequestInfo{
+					Name:         "test-server1",
```

Comment:
wouldn't it be `test-signer2-server1` for consistency?

Reply:
might as well be, I think I was moving code around a lot here so some of the names remained the same :)